cluster/addons/metrics-server/resource-reader.yaml

```diff
@@ -23,7 +24,6 @@ rules:
   verbs:
   - get
   - list
-  - update
```

Comment:
this was added in https://github.com/kubernetes/kubernetes/commit/17c9e344599105b77f38ca7e54aff5dd5e457603#diff-e56c72ed8f82cf9883ae09e465a4675d and while I agree this is overly broad, we need to make sure that the use case that required it is still accomodated

Reply:
cc @piosz

Reply:
I can probably change it to only allow updates to that specific deployment.

Reply:
Yes, you are right.

The nanny side-car needs as permission is to `list` and `watch` for `Nodes`, `get` on `Pod` (we can't know the name in advance), `get` and `update` on `Deployment` (for which we can use the deployment name and `Role`). 
https://github.com/kubernetes/autoscaler/blob/cd56380d76de1587c6c71c6cb62c5dfb3bd1f0e3/addon-resizer/nanny/kubernetes_client.go#L32-L111

I'm not sure how the nanny sidecar is working currently as it it clearly requires `get` to `Pods` [in the namespace](https://github.com/kubernetes/autoscaler/blob/cd56380d76de1587c6c71c6cb62c5dfb3bd1f0e3/addon-resizer/nanny/kubernetes_client.go#L56), but I don't see any bindings giving it access to it.